### DISCUSSION TITLE
v1.57.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#2523] [Announcements]
# v1.57.0

## Highlights

Hello, Immich fans. This release includes improvements to the latest features and solves some long-standing issues. Below are some of the highlights:

### Map improvements for large galleries

The API response is now optimized to return minimal data, which reduces server memory usage and improves network latency.

### Introduce a custom storage label for each user account

The admin can now add a custom label for each user, so instead of `library/{userId}/your-template` it will be `library/{custom_user_label}/your-template`. 

<img width="1392" alt="image" src="https://github.com/immich-app/immich/assets/27055614/c1278715-998d-405c-bc97-18e27584824c">

**NOTE: After setting the template name, you must run the Storage Template Migration job to move files from the older storage path to the new one.**

### Improve video transcoding with new options

Thank @mertalev, for a great PR. We have added additional settings for FFMPEG, including an option to enable two-pass encoding and a better description of previously available options.

<img width="801" alt="image" src="https://github.com/immich-app/immich/assets/27055614/a47b352b-e1b3-4449-bdcd-ce9a8ff542f8">

Along with those features, we have added potential fixes for the microservices container crashing when downloading bad data files for the geocoding process. We also have smoothed out several job queues to avoid hogging the system/CPU. Please help us with your feedback on those improvements.

As always, please consider supporting the project.

🎉 Cheer! 🎉

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="Loading ~4000 images/videos"> 
</p>

If you find the project helpful and it helps you in some ways, you can support the project [one time](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502) or [monthly](https://github.com/sponsors/alextran1502) from GitHub Sponsors

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Mobile
* chore(mobile): use Record instead of custom pair+triple by @fyfrey in https://github.com/immich-app/immich/pull/2483
* chore(mobile): remove hive by @fyfrey in https://github.com/immich-app/immich/pull/2497

### Server
* chore(server): auth request type by @jrasm91 in https://github.com/immich-app/immich/pull/2502
* fix(server): reverse geocoding crash loop by @jrasm91 in https://github.com/immich-app/immich/pull/2489
* chore(server): queue handlers shouldn't increase concurrency by @mertalev in https://github.com/immich-app/immich/pull/2508
* feat(server): add missing thumbnail check to nightly jobs by @jrasm91 in https://github.com/immich-app/immich/pull/2510
* chore(server): remove unused filename by @jrasm91 in https://github.com/immich-app/immich/pull/2517
* chore(server): Remove dist directory in command script by @alextran1502 in https://github.com/immich-app/immich/pull/2518
* feat(server): pagination for asset queries in jobs by @michelheusschen in https://github.com/immich-app/immich/pull/2516
* feat(server): transcode bitrate and thread settings by @mertalev in https://github.com/immich-app/immich/pull/2488

### Web
* feat(web+server): map improvements by @michelheusschen in https://github.com/immich-app/immich/pull/2498
* fix(web): context menu overlap + outclick types by @michelheusschen in https://github.com/immich-app/immich/pull/2506
* feat(web,server): user storage label by @jrasm91 in https://github.com/immich-app/immich/pull/2418
* feat(web): confirm before deleting all faces and people by @jrasm91 in https://github.com/immich-app/immich/pull/2496

### Others
* fix(nginx): enable gzip and show error logs by @michelheusschen in https://github.com/immich-app/immich/pull/2504
* docs: update contributing pages by @jrasm91 in https://github.com/immich-app/immich/pull/2503

## New Contributors
* @mertalev made their first contribution in https://github.com/immich-app/immich/pull/2508

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.56.2...v1.57.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.57.0'>v1.57.0</a>.</em>

**Comment by hlobit** [comment]
Release `#`100 !
Thanks for the awesome piece of work !
